### PR TITLE
Readme install

### PR DESCRIPTION
**Pull request by mitchelljkotler**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,11 +17,42 @@ Analyze, Annotate, Publish. Turn documents into data.
 2. Enter the directory - `cd documentcloud`
 3. Run the dotenv initialization script - `python initialize_dotenvs.py`
    This will create files with the environment variables needed to run the development environment.
-4. Start the docker images - `inv up`
-   This will build and start all of the docker images using docker-compose. It will bind to port 80 on localhost, so you must not have anything else running on port 80. The invoke tasks specify the `local.yml` configuration file for docker-compose. If you would like to run docker-compose commands directly, set the environment variable `export COMPOSE_FILE=local.yml`.
+4. Start the docker images - `export COMPOSE_FILE=local.yml; docker-compose up`
+    -   This will build and start all of the docker images using docker-compose. It will bind to port 80 on localhost, so you must not have anything else running on port 80. The `local.yml` configuration file has the docker-compose details.
```
Comment (mitchelljkotler) [inline README.md:21]: I believe squarelet binds to port 80 not DocumentCloud - this also might need to be moved to after squarelet

File: README.md [hunk 2]
```diff
@@ -17,11 +17,42 @@ Analyze, Annotate, Publish. Turn documents into data.
...
 5. Set `api.dev.documentcloud.org` and `minio.documentcloud.org` to point to localhost - `sudo echo "127.0.0.1 api.dev.documentcloud.org minio.documentcloud.org" >> /etc/hosts`
 6. Enter `api.dev.documentcloud.org/` into your browser - you should see the Django API root page.
-7. Install and run [Squarelet](https://github.com/muckrock/squarelet) and the [DocumentCloud frontend](https://github.com/muckrock/documentcloud-frontend) following the instructions in their repos to view the full-stack application.
+7. Install and run [Squarelet](https://github.com/muckrock/squarelet) and the [DocumentCloud frontend](https://github.com/muckrock/documentcloud-frontend) following the instructions in their repos to view the full-stack application. 
+    - `inv createsuperuser` in Squarelet (not DocumentCloud).
```
Comment (mitchelljkotler) [inline README.md:25]: I believe his should be `inv manage creaturesuperuser`

File: README.md [hunk 3]
```diff
@@ -17,11 +17,42 @@ Analyze, Annotate, Publish. Turn documents into data.
...
+      read_auth_token
+      ```
+   - Add a Client profile and set the `Webhook URL`: `http://api.dev.documentcloud.org/squarelet/webhook/`. The `Source` can remain "MuckRock" in the dropdown.
+   - You will have to look up **three** security-related configuration lines from your Squarelet instance to DocumentCloud (as if it was MuckRock). 
+   - Write their values into the `./.envs/.local/.django` file of the DocumentCloud repository, which should already be initialized from above.
+   - `Client ID` goes into `SQUARELET_KEY`
+   - `Client Secret` goes into `SQUARELET_SECRET`
+   - Additionally, get the value for `JWT_VERIFYING_KEY` by opening the Squarelet Django shell using `inv shell` and copying the `settings.SIMPLE_JWT['VERIFYING_KEY']` (remove the leading `b'` and the trailing `'`, leave the `\n` portions as-is)
+9. Set `SIDEKICK_PROCESSING_URL=mock://sidekick.dev.documentcloud.org`
+10. Set `SQUARELET_WHITELIST_VERIFIED_JOURNALISTS=false` to allow you to login.
```
Comment (mitchelljkotler) [inline README.md:50]: These two can be added to `initialize_dotenvs.py`

File: README.md (hunk 1 above)
Comment (thadk) [reply README.md:22]: I forgot about Nginx at the center of the Docker squarelet network on 127.0.0.1. I think it gets a mention elsewhere. It would be good to document it explicitly here, and I'll try to. I went digging in the Dockerfiles and noticed that this custom image is used locally instead of python:3.6-alpine used on production –  do you remember why? https://hub.docker.com/r/matthewfeickert/docker-python3-ubuntu/ /  https://github.com/matthewfeickert/Docker-Python3-Ubuntu
This will slow performance on any M1 Macs since unlike the official image there is no amd64 version.

File: README.md (hunk 2 above)
Comment (thadk) [reply README.md]: I believe this is resolved on the latest commit.

File: README.md (hunk 3 above)
Comment (thadk) [reply README.md]: I believe this has been resolved on the latest commit

File: README.md [hunk 4]
```diff
@@ -13,15 +13,47 @@ Analyze, Annotate, Publish. Turn documents into data.
...
 1. Check out the git repository - `git clone git@github.com:MuckRock/documentcloud.git`
 2. Enter the directory - `cd documentcloud`
 3. Run the dotenv initialization script - `python initialize_dotenvs.py`
    This will create files with the environment variables needed to run the development environment.
-4. Start the docker images - `inv up`
-   This will build and start all of the docker images using docker-compose. It will bind to port 80 on localhost, so you must not have anything else running on port 80. The invoke tasks specify the `local.yml` configuration file for docker-compose. If you would like to run docker-compose commands directly, set the environment variable `export COMPOSE_FILE=local.yml`.
+4. Start the docker images - `export COMPOSE_FILE=local.yml; docker-compose up`
+    -   This will build and start all of the DocumentCloud docker images using docker-compose. It will attach to the Squarelet network which must be already running. You can connect to Squarelet nginx on port 80 and it will serve the pages based on domain as a virtual host. The `local.yml` configuration file has the docker-compose details.
```
Comment (thadk) [inline README.md:22]: Here's the latest language to review

File: README.md [hunk 5]
```diff
@@ -13,15 +13,47 @@ Analyze, Annotate, Publish. Turn documents into data.
...
 1. Check out the git repository - `git clone git@github.com:MuckRock/documentcloud.git`
 2. Enter the directory - `cd documentcloud`
 3. Run the dotenv initialization script - `python initialize_dotenvs.py`
    This will create files with the environment variables needed to run the development environment.
-4. Start the docker images - `inv up`
-   This will build and start all of the docker images using docker-compose. It will bind to port 80 on localhost, so you must not have anything else running on port 80. The invoke tasks specify the `local.yml` configuration file for docker-compose. If you would like to run docker-compose commands directly, set the environment variable `export COMPOSE_FILE=local.yml`.
+4. Start the docker images - `export COMPOSE_FILE=local.yml; docker-compose up`
+    -   This will build and start all of the DocumentCloud docker images using docker-compose. It will attach to the Squarelet network which must be already running. You can connect to Squarelet nginx on port 80 and it will serve the appropriate dependent http service, such as DocumentCloud, based on domain as a virtual host. The `local.yml` configuration file has the docker-compose details.
```
Comment (thadk) [inline README.md:22]: Here's the latest language to review and the subsequent line

File: README.md (hunk 1 above)
Comment (mitchelljkotler) [reply README.md]: I believe either pdfium or tesseract (or possibly both) did not run under alpine, so an ubuntu image was used.

Comment (thadk) [reply README.md]: Maybe `3.6-bullseye` (Debian latest), `3.6-slim-bullseye` or `3.6-buster` (Debian LTS) could be used instead? It shows them at https://hub.docker.com/_/python/

Comment (mitchelljkotler) [reply README.md]: Do you use an M1 Mac?  I don't think its worth messing around with the base image if its not currently affecting anyone

**Comment by thadk** [comment]
(Note to myself: I guess we should probably also mention that this is the installation of the local development environment for DocumentCloud.)

File: README.md [hunk 6]
```diff
@@ -11,17 +11,78 @@ Analyze, Annotate, Publish. Turn documents into data.
...
-7. Install and run [Squarelet](https://github.com/muckrock/squarelet) and the [DocumentCloud frontend](https://github.com/muckrock/documentcloud-frontend) following the instructions in their repos to view the full-stack application.
+5. You do not need to immediately start the docker-compose session for DocumentCloud until the Squarelet authentication service integration below is complete.
+6. Set `api.dev.documentcloud.org` and `minio.documentcloud.org` to point to localhost - `sudo echo "127.0.0.1 api.dev.documentcloud.org minio.documentcloud.org" >> /etc/hosts`
+7. Enter `api.dev.documentcloud.org/` into your browser - you should see the Django API root page. **Note that `api` is before `dev` in this service URL.**
+8. Install and run [Squarelet](https://github.com/muckrock/squarelet) and the [DocumentCloud frontend](https://github.com/muckrock/documentcloud-frontend) following the instructions in their 2 repos as part of the full-stack application. 
+    - **Create Squarelet user:** 
+      - `inv sh` (not shell) then `./manage.py createsuperuser` in Squarelet (not DocumentCloud).
```
Comment (mitchelljkotler) [inline README.md:29]: These instructions should probably be in the Squarelet readme - also I think it is simpler to do `inv manage createsuperuser`

File: README.md [hunk 7]
```diff
@@ -11,17 +11,78 @@ Analyze, Annotate, Publish. Turn documents into data.
...
+      read_user
+      read_organization
+      read_auth_token
+      ```
+   - Add a Client profile and set the `Webhook URL`: `http://api.dev.documentcloud.org/squarelet/webhook/`. The `Source` can remain "MuckRock" in the dropdown.
+   - You will have to look up **three** authentication-related configuration lines from your Squarelet instance to insert into DocumentCloud `.django` file (as if it was MuckRock). 
+   - Write their values into the `./.envs/.local/.django` file of the DocumentCloud repository, which should already be initialized from above.
+   - `Client ID` goes into `SQUARELET_KEY`
+   - `Client Secret` goes into `SQUARELET_SECRET`
+   - Additionally, get the value for `JWT_VERIFYING_KEY` by opening the Squarelet Django shell using `inv shell` and copying the `settings.SIMPLE_JWT['VERIFYING_KEY']` (remove the leading `b'` and the trailing `'`, leave the `\n` portions as-is)
```
Comment (mitchelljkotler) [inline README.md:52]: It would be nice to create an invoke task which automates this process

File: README.md [hunk 8]
```diff
@@ -11,17 +11,78 @@ Analyze, Annotate, Publish. Turn documents into data.
...
+   - (If `JWT_VERIFYING_KEY` is blank, don't forget to `inv sh` on Squarelet and then run `./manage.py creatersakey` as the instructions linked above explained)
+10. In Squarelet `.django` file, you may need to provide valid testing values for `STRIPE_PUB_KEYS`, `STRIPE_SECRET_KEYS` and set `STRIPE_WEBHOOK_SECRETS=None` from the MuckRock team (multiple values are comma separated only, no square braces)
+      - You must always fully `docker-compose down` or Ctrl-C each time you change a `.django` file of a docker-compose session for it to take effect (as far as I know).
```
Comment (mitchelljkotler) [inline README.md:55]: Same comments as in Squarelet repo

File: README.md [hunk 9]
```diff
@@ -11,17 +11,78 @@ Analyze, Annotate, Publish. Turn documents into data.
...
+    - If you do `docker-compose down` on Squarelet when none of the other dependent docker-compose sessions (such as DocumentCloud) are running, `docker-compose down` will delete the Squarelet network. You will have to explicitly bring the whole squarelet docker-compose session back up to recreate it and nginx before being able to start a dependent docker-compose session (such as DocumentCloud).
+    - Using `docker-compose up -d` rather than `docker-compose up` will make a daemon for DocumentCloud as Squarelet defaults to.
+13. Log in using the Squarelet superuser on the locally-running [Documentcloud-frontend](https://github.com/muckrock/documentcloud-frontend) that you installed earlier at http://dev.documentcloud.org
+    - If it works, close the window and continue. Do not try to upload any documents yet. 
+    - If it does not work, check your authentication integration settings. This may take several tries. `SQUARELET_WHITELIST_VERIFIED_JOURNALISTS=True` environment variable makes it so only verified journalists can *log into* DocumentCloud.
```
Comment (mitchelljkotler) [inline README.md:64]: This should not take several tries - the instructions should specify setting the env vars in the correct order so it works the first time.

File: README.md [hunk 10]
```diff
@@ -11,17 +11,78 @@ Analyze, Annotate, Publish. Turn documents into data.
...
+    - Use the squarelet admin [Organization page](http://dev.squarelet.local/admin/organizations/organization/) to mark your organization as a verified journalist to allow upload to DocumentCloud.
+    - **Make your Squarelet superuser also a superuser on DocumentCloud Django:** Run `inv shell` in the DocumentCloud folder and use these commands (no indent):
+      ```
+      tempUser = User.objects.all()[0]
+      tempUser.is_superuser = True
+      tempUser.save()
+      ```
+14. If you can login successfully, go to [Django admin for DocumentCloud](http://api.dev.documentcloud.org/admin) and add the required static [flat page](http://api.dev.documentcloud.org/admin/flatpages/flatpage/) called `/tipofday/`. It can be blank. Do not prefix the URL with `/pages/`. Specifying the `Site` as `example.com` is alright.
```
Comment (mitchelljkotler) [inline README.md:72]: Remove if you can login successfully - if they followed the instructions they should be able to

File: README.md (hunk 7 above)
Comment (thadk) [reply README.md:52]: From conversation with Mitch: Note that JWT piece is DocumentCloud specific and not in Muckrock codebase: it uses a different token type.

File: README.md (hunk 6 above)
Comment (thadk) [reply README.md:29]: Fixed in next commit

File: README.md (hunk 9 above)
Comment (thadk) [reply README.md:64]: Fixed

File: README.md (hunk 10 above)
Comment (thadk) [reply README.md:72]: Fixed

File: README.md (hunk 2 above)
Comment (thadk) [reply README.md]: Actually, we need to revert this to what was in the last version:
```
stall ✗)]$ /Users/thadk/Library/Python/3.8/bin/inv manage createsuperuser
Container squarelet_squarelet_postgres_1  Running
Container squarelet_squarelet_mailhog_1  Running
Container squarelet_squarelet_nginx_1  Running
PostgreSQL is available
Superuser creation skipped due to not running in a TTY. You can run `manage.py createsuperuser` in your project to create one manually.
```
